### PR TITLE
HHH-9765 - Fix JPA Exception Mapping during flushing SyncronizationCallb...

### PR DESCRIPTION
**Pull request by dreab8**
...ack beforeCompletion()

https://hibernate.atlassian.net/browse/HHH-9765

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/resource/transaction/TransactionCoordinator.java [hunk 1]
```diff
@@ -113,6 +114,8 @@
 
 	public int getTimeOut();
 
+	public void setExceptionMapper(ExceptionMapper exceptionMapper);
+
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/resource/transaction/TransactionCoordinator.java]: I'm not overly a fan of setting the `ExceptionMapper`.  I'd much rather see this handled via something like a new `org.hibernate.engine.spi.SessionOwner` method, or another means of injecting this into the Session itself.

**Comment by sebersole** [comment]
Also, we seemed to have missed `ManagedFlushChecker` and `AfterCompletionAction` as well.  Or are those handled differently somehow today?

**Comment by dreab8** [comment]
yes we have to add also ManagedFlushChecker and AfterCompletionAction, I'm going to do this.

File: hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java [hunk 2]
```diff
@@ -2180,6 +2218,9 @@ private void writeObject(ObjectOutputStream oos) throws IOException {
 
 		oos.defaultWriteObject();
 
+		oos.writeObject( exceptionMapper );
+		oos.writeObject( managedFlushChecker );
+		oos.writeObject( afterCompletionAction );
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java]: we already write the SessionOwner to the stream later.  Writing these is completely unnecessary.

**Comment by sebersole** [comment]
This was applied before to master, but forgot to close the PR.  Thanks Andrea!

**Pull request closed without merging**